store/index.js

```diff
@@ -1,6 +1,45 @@
 import { defineStore } from 'pinia';
 import axios from 'axios';
 
+const monsterFields = [
```

Comment:
The lion's share of new functionality is in here. This has been almost completely rewritten to streamline API requests and manage the source selection.